### ISSUE TITLE
Issue with setting up local environment

### ISSUE THREAD

**Issue by rahul1990gupta** [freeCodeCamp/chapter#416]
### Describe the bug:

There are few issues with the local dev environment
- `await initDB()` fails silently without starting a server. If I replace the code with the snippet below, it works fine. 
```js
  initDB() 
  .then(() => {
    console.log("connected");
  }).catch((err) => {
    console.log(err);
  });
```
- `yarn both` gives me the following error 
```
$ yarn both
yarn run v1.22.5
$ concurrently "yarn dev" "yarn --cwd='./client' dev"
$ ts-node-dev --no-notify -P tsconfig.server.json ./server/app.ts
$ ts-node-dev --no-notify -P tsconfig.server.json ./server/app.ts
[0] ts-node-dev ver. 1.0.0-pre.63 (using ts-node ver. 8.10.2, typescript ver. 3.9.7)
[1] ts-node-dev ver. 1.0.0-pre.63 (using ts-node ver. 8.10.2, typescript ver. 3.9.7)
[1] in module
[0] in module
[1] After express
[0] After express
[0] DB initialized
[1] DB initialized
[0] built schema
[1] built schema
[0] Created server
[0] applied middleware
[0] prepared options.
[1] Created server
[1] applied middleware
[1] prepared options.
[0] Listening on http://localhost:5000/graphql
[1] Error: listen EADDRINUSE: address already in use :::5000
[1]     at Server.setupListenHandle [as _listen2] (net.js:1317:16)
[1]     at listenInCluster (net.js:1365:12)
[1]     at Server.listen (net.js:1451:7)
[1]     at Function.listen (D:\repos\chapter\node_modules\express\lib\application.js
:618:24)
[1]     at D:\repos\chapter\server\app.ts:62:9
[1]     at processTicksAndRejections (internal/process/task_queues.js:93:5)
[1] [ERROR] 19:53:59 Error: listen EADDRINUSE: address already in use :::5000

``` 
On the other hand, running both the servers in seperate terminals worked fine for me.
### Tell us about your browser and operating system:
- Browser(s) name and version: 
- Operating System: Windows 10 Home edition 
- Node Version: 14.13.10

**Comment by Ryuno-Ki** [comment]
Since it came up on Discord: Those messages were added for debugging by the OP.
It's a setup without Docker.

**Comment by timmyichen** [comment]
> `await initDB()` fails silently without starting a server. If I replace the code with the snippet below, it works fine.

This should be blocking for the rest of the main function, because later operations depend on the server connection existing.

Can you tell us more about what "working fine" means?  Your additions provide more accurate reporting in the console, but wouldn't actually fix anything that broke.

**Comment by rahul1990gupta** [comment]
@timmyichen 
With `await InitDB()`, `yarn dev` server exits before initializing DB properly. 
As you can see in the screenshot below, I have put in some console.log in the left part and output is shown in the right. Program exits in 6 seconds without spawning a server.

But if I replace it with `.then().catch()` block it executes all the following steps in order and spawns a apollo graphql server to listen  at port 5000.

![image](https://user-images.githubusercontent.com/8749679/96664918-657f3600-1371-11eb-81f6-15a97969e1b1.png)

**Comment by rahul1990gupta** [comment]
When I switch node version to 12.19.0, I can start the backend server without having the program exit. [My guess is node 13 will work too. I didn't try it because npm complained about node 13 not being a stable version]

**Comment by allella** [comment]
@rahul1990gupta please check out #417 and see if this is a fix.

**Comment by rahul1990gupta** [comment]
@allella The fix works for me on Node `14.13.0`. Thanks.

**Comment by allella** [comment]
@all-contributors add @rahul1990gupta for bug
